### ISSUE TITLE
`reportUninitializedInstanceVariable` false positives for `TypedDict`

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#3441]
**Describe the bug**
A recent update is now telling me that members of a TypedDict are not initialized, despite TypedDicts requirering all members to be given as constructor arguments. Pyright itself reports that `TypedDict classes can contain only type annotations`. Meaning that one way or the other. TypedDict can never pass type checking.

**To Reproduce**
See code snippet

**Expected behavior**
In the snippet below, only the last two lines should be invalid. The TypedDict class definition should be fine. 

**Screenshots or Code**
```py
from typing import TypedDict
class MyDict(TypedDict):
    member1: str  # Error, not expected
    member2: str  # Error, not expected


MyDict(member1="", member2="")  # No error, as expected
MyDict(member1="")  # No error, as expected
MyDict("", "")  # No error, as expected
```
![image](https://user-images.githubusercontent.com/1350584/167274901-fc692ea8-d963-429e-8e6f-f18b440b56f0.png)
```
C:\example.py
  C:\example.py:30:5 - error: Instance variable "member1" is not initialized in the class body or __init__ method (reportUninitializedInstanceVariable)
  C:\example.py:31:5 - error: Instance variable "member2" is not initialized in the class body or __init__ method (reportUninitializedInstanceVariable)
  C:\example.py:35:1 - error: Argument missing for parameter "member2" (reportGeneralTypeIssues)
  C:\example.py:36:8 - error: Expected 0 positional arguments (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Command line pyright@1.1.245

**Additional context**
Sounds similar to #3431

**Comment by erictraut** [comment]
Yeah, `TypedDict` classes need to be special-cased here since class variables declared within a `TypedDict` are not actually class variables. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.246, which I just published. It will also be included in the next release of pylance.
